AUTHORS: Update authors from git

gxalpha opened the pull request:
<!--- Please fill out the following template, which will help other contributors review your Pull Request. -->

<!--- Make sure you’ve read the contribution guidelines here: https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst -->

### Description
<!--- Describe your changes in detail. -->
<!--- If this change includes UI elements, please include screenshots. -->
Updates the git authors to the state of 8ce58b0 (also updates the .mailmap file, see the conversation).
Will show on the about page.

Created with
```
git shortlog -sn --no-merges
```
and removed the commit numbers.

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open GitHub Issue, or implements feature request -->
<!--- from the Ideas page, please link to the issue here. -->
The git authors list hasn't been updated since January 2019, just the translators.
When I asked about this on the discord, Fenrir agreed it might be due to an update.
He noted that last time this was done, some people didn't want to be on there but we concluded that since this information is publicly available, that probably shouldn't be a problem.
I see that this could be debatable though.

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment (hardware, OS version, etc.),-->
<!--- and the tests you ran, including how it may affect other areas of code. -->
Compiled on my machine (macOS Big Sur 11.0.1, Intel) and it showed on the authors list in the about tab.

### Types of changes
<!--- What types of changes does your PR introduce? Uncomment all that apply -->
<!--- - Bug fix (non-breaking change which fixes an issue) -->
<!--- - New feature (non-breaking change which adds functionality) -->
<!--- - Performance enhancement (non-breaking change which improves efficiency) -->
- Code cleanup (non-breaking change which makes code smaller or more readable)
<!--- - Breaking change (fix or feature that would cause existing functionality to change) -->
- Documentation (a change to documentation pages)

(Probably documentation, not exactly sure where to put it)
(Could be cleanup as well i guess)

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

Xaymar commented on AUTHORS:
```diff
@@ -5,211 +5,429 @@ Contributors are sorted by their amount of commits / translated strings.
-Xaymar
+derrod
+Kurt Kartaltepe
+Michael Fabian Dirks
```
I'm on this list thrice.

Xaymar commented on AUTHORS:
```diff
@@ -5,211 +5,429 @@ Contributors are sorted by their amount of commits / translated strings.
+VodBox
 SuslikV
-Rodney
+Michael Fabian 'Xaymar' Dirks
```
2nd Instance (this one is correct though)

Xaymar commented on AUTHORS:
```diff
@@ -5,211 +5,429 @@ Contributors are sorted by their amount of commits / translated strings.
+Will Jamieson
+William Casarin
+Wouter
+Xaymar
```
3rd instance

gxalpha commented:
Good find, I didn't think about people using different git names for their commits (which is probably what you did).
Do you think using the email addresses used for the commits would be better to find out who is the same person (and then combine that information)?

I think I'm going to try that and mark this as draft in the meantime.

Xaymar commented:
There's a .mailmap file for that exact purpose if I remember correctly.

gxalpha commented:
Yup, and from what I understand, it should combine everything with `git shortlog -sne`(?). 
It doesn't.

I'll take a look at tomorrow, probably overlooked something obvious.

gxalpha commented:
So it turns out the mailmap file does work, the current configuration of your entry though just replaces your old email with the new one. I got it to work by removing your old email address from the file; but that then left two entries, the new one with the new email address (with all the commits and the correct name) and the one with the old email address, with the old commits. 

This is what I came up with, and while it solved the problem, I am not happy with how it looks.

Xaymar commented on .mailmap:
```diff
@@ -4,5 +4,8 @@ Benjamin Klettbach <b.klettbach@gmail.com>
-Michael Fabian Dirks <info@xaymar.com> <michael.dirks@xaymar.com>
+Michael Fabian Dirks <info@xaymar.com> <Xaymar@users.noreply.github.com>
+Michael Fabian Dirks <info@xaymar.com> <michael.dirks@xaymar.com> 
+Michael Fabian Dirks <info@xaymar.com>
```
The credited name for me should be `Michael Fabian 'Xaymar' Dirks`, not just my legal name.

Xaymar commented:
Seems like that is how the .mailmap format is supposed to work. Map one e-mail to another. I'm not actually sure why the github-noreply one is in there, since I've never actually set Github to use that.

gxalpha commented:
There was like one commit in there with that email, so I decided that if I'm gonna do it i should probably add that as well.
I can also change thename, sure. I just left it like it was before.

Xaymar commented:
`Michael Fabian 'Xaymar' Dirks` is my registered artist name, so when it comes to crediting work it should be with the artist name. Thanks for doing this by the way.

gxalpha commented:
Ok fixed that.
Also: I think two commits are appropriate since the module name should be different, it would just look weird to have a commit starting with "AUTHORS" to be the most recent commit of the mailmap file, for potentially years (and that would be on the top-level of the repo).
Marking this as review-ready again, feel free to review :D

Programatic commented on AUTHORS:
```diff
@@ -5,211 +5,428 @@ Contributors are sorted by their amount of commits / translated strings.
 nleseul
+David Bieber
+Dead133
+Programatic
```
This is the second time i'm on here

gxalpha commented:
@Programatic what do you want to be called? Your real name, your nickname or something in between?

Programatic commented:
Sorry, I should have specified. My real name is Ford Smith which is already in there

gxalpha commented:
Ok, so:
I made a small script finding (hopefully) all of the duplicates and creating the necessary parts of the mailmap file which i added to the mailmap file.
From that I created the authors list.

Everyone that has matching emails but different names or vice versa should only be on there once now. Obviously if you used a different email *and* name that will not be detected, please reach out and I'll add you.

Also kinda funny how i managed to mess with the git commits in a way that now the commit date is a week ago, lol

gxalpha commented:
Next update:

After being told that people should probably keep the names they had before, I changed the name of all of those had a double digit number of commits (only those because I had to find them out myself) to what the nickname is on either GitHub or Discord and added it to the mailmap.

This is starting to more and more become a mailmap-change instead of an authors-change, but personally I don't like the idea of just changing the names in the authors-list since if this gets done at some point in the future again, doing this would be way easier since changing the authors list is literally generating it with `git shortlog -sn --no-merges`, removing the commit numbers and then a simple copy-paste.

gxalpha commented:
This is kinda stall right now, so I updated it again.
Hope its not gonna end up like the other PRs that are open for months and nothing happens, so if there is anything else required, please tell me

Fenrirthviti commented:
We still only have one full-time developer who is taking time off for the holidays, and we just had a release. That kind of comment is entirely unnecessary and, frankly, quite rude. Please refrain in the future.

This will be looked at when time allows.

gxalpha commented:
You're right, I kinda wrote this message in a hurry and didn't think too much about it.

I'm sorry.
